src/ui/mod.rs

```diff
@@ -99,4 +99,45 @@ mod test {
         assert_eq!(colors.len(), 2);
         assert_eq!(colors, vec![num_to_color(&2), num_to_color(&3)]);
     }
+
+    #[test]
+    fn get_ascii_colors_fill_custom_colors_with_language_colors() {
+        // When custom ascii colors are not enough for the given language,
+        // language colors should be used as default
+        let colors = get_ascii_colors(&None, &Language::Go, &[0], false);
+        assert_eq!(colors.len(), 3);
+        assert_eq!(
+            colors,
+            vec![
+                num_to_color(&0),
+                DynColors::Ansi(AnsiColors::Default),
+                DynColors::Ansi(AnsiColors::Yellow)
+            ]
+        );
+    }
+
+    #[test]
+    fn num_to_color_code_match() {
+        // This test case is only for test coverage
```

Comment:
Not a big fan of test coverage for the sake of task coverage, this is a perfect illustration of [Goodhart’s law](https://en.wikipedia.org/wiki/Goodhart%27s_law)